### PR TITLE
Add e2e tests approval for all for erc1155 token

### LINKED ISSUE #17933: erc1155 setApprovalForAll e2e tests
See: https://github.com/MetaMask/test-dapp/pull/215

### PR DESCRIPTION
**Pull request by chloeYue**
## Explanation
 
- Fix #17933 : In this PR, 2 e2e testcase are added for set approval and revoke approval for erc1155 token
- Reorganization of nft tests
- Stabilize tests

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by PeterYinusa — DISMISSED** [review]

File: test/e2e/nft/erc1155-interaction.spec.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        );
+
+        // Confirm Mint
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+        await driver.clickElement('[data-testid="home__activity-tab"]');
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .transaction-list-item:nth-of-type(1)',
+          { timeout: 10000 },
```
Comment (seaona) [inline test/e2e/nft/erc1155-interaction.spec.js:61]: I think we could remove this timeout as it's already set to `10000` on the `waitForSelector` method

File: test/e2e/nft/erc1155-interaction.spec.js [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        );
```
Comment (seaona) [inline test/e2e/nft/erc1155-interaction.spec.js:62]: maybe finishing  the tests with an assertion could be interesting, so we can also add a readable message if the assertion fails

**Comment by seaona** [comment]
Overall looks good! Nice coverage! Just left some small suggestions, not a big deal ,take it or leave it :)

**Review by seaona — DISMISSED** [review]

File: test/e2e/nft/erc1155-interaction.spec.js (hunk 1 above)
Comment (chloeYue) [reply test/e2e/nft/erc1155-interaction.spec.js]: Done, thanks !

File: test/e2e/nft/erc1155-interaction.spec.js (hunk 2 above)
Comment (chloeYue) [reply test/e2e/nft/erc1155-interaction.spec.js:67]: Done, thanks !

**Review by PeterYinusa — APPROVED** [review]

**Review by seaona — APPROVED** [review]

**Pull request merged**